### PR TITLE
Devices enumeration: workaround for MacOS

### PR DESCRIPTION
**Pull request by andreibancioiu**
On MacOS >= 13.3.1, device enumeration produces more records, e.g.

```
hid_device {'path': b'DevSrvsID:4294970322', 'vendor_id': 11415, 'product_id': 4117, 'serial_number': '0001', 'release_number': 513, 'manufacturer_string': 'Ledger', 'product_string': 'Nano S', 'usage_page': 65440, 'usage': 1, 'interface_number': 0}
hid_device {'path': b'DevSrvsID:4294970324', 'vendor_id': 11415, 'product_id': 4117, 'serial_number': '0001', 'release_number': 513, 'manufacturer_string': 'Ledger', 'product_string': 'Nano S', 'usage_page': 61904, 'usage': 1, 'interface_number': 0}
```
The order of the enumeration is not deterministic, and, sometimes, `HID.enumerate_devices(self.vendor_id)[0]` might pick an unusable record (not the actual device). In this PR, we implement the device picking logic as a new function, `_decide_device_path`, which handles the behavior of MacOS >= 13.3.1, as well.

The logic of `HID.enumerate_devices()` is left untouched - in case downstream applications were actually relying on the results being a collection of paths, instead of a single path.

## How to test

Clone the repository and, while having a device connected  (with the MultiversX app open - as an example), run the following command:

```
python3 -m ledgercomm.cli.send --hid stdin <<<ed02000000
```

It should work with no error (on all operating systems). Output should be similar to:

```
ledgercomm - DEBUG - => ed02000000
ledgercomm - DEBUG - <= 0102000100140000000000000000 9000
```

Extra references:
 - https://developer.apple.com/forums/thread/728001
 - https://github.com/multiversx/mx-sdk-py-cli/issues/253

### CODE REVIEW

File: ledgercomm/interfaces/hid_device.py [hunk 1]
```diff
@@ -35,8 +43,7 @@ class HID(Comm):
     def __init__(self, vendor_id: int = 0x2C97) -> None:
```
Comment (fbeutin-ledger) [inline ledgercomm/interfaces/hid_device.py]: Use the new DEFAULT_VENDOR_ID

Comment (andreibancioiu) [reply ledgercomm/interfaces/hid_device.py]: Thank you, fixed.

**Comment by lpascal-ledger** [comment]
Thanks for the PR @andreibancioiu! We'll need some time to test it on a proper device, but it'll definitely be integrated then.

File: ledgercomm/interfaces/hid_device.py [hunk 2]
```diff
@@ -53,13 +63,45 @@ def open(self) -> None:
...
+
+        devices = [device for device in devices if device.get("interface_number") == 0]
+        if len(devices) == 1:
+            # No ambiguity, pick the only device
+            return devices[0]["path"]
+
+        # On MacOS, "interface_number" is not a reliable filtering criteria, so we
+        # also filter by "usage_page".
+        devices = [device for device in devices if device.get(
+            "usage_page") == MACOS_USAGE_PAGE]
```
Comment (fbeutin-ledger) [inline ledgercomm/interfaces/hid_device.py]: Hello, this fails on Linux if two devices are plugged. I've fixed it with:
```suggestion
        mac_filtered_devices = [device for device in devices if device.get(
             "usage_page") == MACOS_USAGE_PAGE]
        if len(mac_filtered_devices) != 0:
            # If we are on a MAC device, we keep the new filtered list
            devices = mac_filtered_devices
```

**Comment by fbeutin-ledger** [comment]
Hello @andreibancioiu , sorry about this the CI is failing.. Can you take a quick look please? I'll do it next week otherwise

**Comment by andreibancioiu** [comment]
> Hello @andreibancioiu , sorry about this the CI is failing.. Can you take a quick look please? I'll do it next week otherwise

Hello @fbeutin-ledger,

Thanks for the note :pray: 

Fixed the linting issues:
 - I was using the deprecated `LOG.warn()` instead `LOG.warning()`
 - I was raising the base `Exception` at some point. Fixed by defining a custom error: `CannotFindDeviceError`

**Comment by fbeutin-ledger** [comment]
Merged into LedgerHQ:fbe/merge_mac_support
I'll fix the lint there

**Comment by fbeutin-ledger** [comment]
https://github.com/LedgerHQ/ledgercomm/pull/14

**Pull request merged**